### ISSUE TITLE
Python: Updating look changes appearence of visualization (even though vis_config doesn't change)

### ISSUE THREAD

**Issue by jpkell05** [looker-open-source/sdk-codegen#214]
For some context, I was planning on using the sdk to update the filters in a look and retrieve the visualization. But, when I update the look (even if I don't change anything) the appearance of some aspects of the pie chart visualization reset to default (specifically the colors used and the orientation). Other aspects do not, for example the inner radius I set does not revert to zero. 

Here you can see the visualization as I set it up. 

![image](https://user-images.githubusercontent.com/15281417/81200194-3f173b00-8f89-11ea-8cfb-207d4948d361.png)

Here I just use the look's info to update it, I'm not changing anything. I check the vis_config after updating and it is the same as before. But when I retrieve the visualization it appears different.

![image](https://user-images.githubusercontent.com/15281417/81200476-9ddcb480-8f89-11ea-88f3-438582d1742f.png)

Furthermore if I navigate to that look on my browser. The visualization appears different there too.

**Comment by joeldodge79** [comment]
thanks for the report. at first blush it seems more likely an issue with our Looker API as opposed to the python sdk in particular.

Was the look (2395) initially created via the api/sdk or in the UI? I'm suspecting there's something wonky with how the vis_config sticks internally when set via API

**Comment by jpkell05** [comment]
I initially created the look in the UI

**Comment by joeldodge79** [comment]
thanks. I'll play around with it and let you know what I find

**Comment by joeldodge79** [comment]
this is definitely specific to the Python SDK:

Query.vis_config in the oas spec is
```
          "vis_config": {
            "type": "object",
            "additionalProperties": {
              "type": "string",
              "format": "any"
            },
```

but our code generator converts that into a `MutableMapping[str,str]` type in python. In the python SDK we deserialize based on the type annotation so every value gets slammed into a string and you end up with something like

```python
{
    'value_labels': 'legend',
    'label_type': 'labPer',
    'inner_radius': '50',
    'color_application': "{'collection_id': 'legacy', 'palette_id': 'mixed_pastels', 'options': {'steps': 5}}",
    'series_colors': '{}',
    'type': 'looker_pie',
    'series_type': '{}',
    'defaults_version': '1',
    'series_types': '{}',
    'show_value_labels': 'True',
    'font_size': '12',
    'up_color': '#3EB0D5',
    'down_color': '#B1399E',
    'total_color': '#C2DD67',
    'show_x_axis_ticks': 'True',
    'show_x_axis_label': 'True',
    'x_axis_scale': 'auto',
    'show_y_axis_labels': 'True',
    'show_y_axis_ticks': 'True',
    'y_axis_gridlines': 'True'
}
```

when you should get
```python
{
    "value_labels": "legend",
    "label_type": "labPer",
    "inner_radius": 50,
    "color_application": {
        "collection_id": "legacy",
        "palette_id": "mixed_pastels",
        "options": {
            "steps": 5
        }
    },
    "series_colors": {},
    "type": "looker_pie",
    "series_type": {},
    "defaults_version": 1,
    "series_types": {},
    "show_value_labels": true,
    "font_size": 12,
    "up_color": "#3EB0D5",
    "down_color": "#B1399E",
    "total_color": "#C2DD67",
    "show_x_axis_ticks": true,
    "show_x_axis_label": true,
    "x_axis_scale": "auto",
    "show_y_axis_labels": true,
    "show_y_axis_ticks": true,
    "y_axis_gridlines": true
}
```

I'll think about a solution. Probably a more accurate type annotation but also need to consider how the `cattr` library we're using for [de]serialization will handle that.

**Comment by jpkell05** [comment]
That's interesting, thanks for the update!

**Comment by joeldodge79** [comment]
I haven't tested this yet but there may be a nasty work around where you can manually reset the values in the vis_config dict to be their "native" type.

e.g.
```python
color_app = json.loads(new_query.vis_config["color_application"])
color_app["options"]["steps"] = int(color_app["options"]["steps"])
new_query.vis_config["color_application"] = color_app
new_query.vis_config["inner_radius"] = int(new_query.vis_config["inner_radius"])
new_query.vis_config["show_x_axis_ticks"] = True if new_query.vis_config["show_x_axis_ticks"] == "True" else False
.
.
.
```
I don't know which ones you'd really need to "reset" in order for it to work but you can play around with it. Hopefully we'll have a good solution soon

**Comment by joeldodge79** [comment]
this issue has been fixed in the latest release (0.1.3b14) and should work out of the box now

**Comment by jpkell05** [comment]
Amazing! Thanks for the quick fix Joel.
